src/osyris/plot/wrappers.py

```diff
@@ -126,3 +129,22 @@ def scatter(ax, x, y, data, **kwargs):
         return coll
     else:
         return ax.scatter(x, y, **default_args)
+
+def line_integral_convolution(ax, x, y, z, **kwargs):
+    
+    lic_res = lic.lic(z[...,1],z[...,0],length=30) #compute line integral convolution
+    
+    #amplify contrast on lic
+    lim=(.2,.5)
+    lic_data_clip = np.clip(lic_res,lim[0],lim[1])
+    lic_data_rgba = ScalarMappable(norm=None, cmap="binary").to_rgba(lic_data_clip)
+    lic_data_clip_rescale = (lic_data_clip-lim[0])/(lim[1]-lim[0])
+    lic_data_rgba[...,3] = lic_data_clip_rescale * 1
+    
+    args = [lic_data_rgba]
+    plot_args = {**kwargs}
+    plot_args["cmap"] = "binary"
+    plot_args["extent"] = [np.min(x),np.max(x),np.min(y),np.max(y)]
+    plot_args["origin"] = "lower"
+    
+    return ax.imshow(*args, **plot_args)
```

Comment:
I was able to play with the colors of `pcolormesh`, but in a non-standard way.
Setting the `alpha` to something else than 1 on a `pcolormesh` is not great because some edge artifacts start to appear (see e.g. https://stackoverflow.com/questions/52100747/changing-facecolor-alpha-value-gives-unwanted-edges-matplotlib-pcolormesh)
So we want to play with the facecolors directly.

I can do something like
```Py
p = pcolormesh(ax=ax,
               x=x,
               y=y,
               z=z[..., 2],
               **kwargs)
# pcolormesh needs a draw call before the array of facecolors is defined
ax.get_figure().canvas.draw()
facecolors = p.get_facecolors()
flat_shape = facecolors.shape[0]
newcolors = facecolors.reshape(*z[..., 2].shape, 4)
newcolors[..., :3] *= 0.6 * (lic_res.reshape(*lic_res.shape, 1) - 0.5) + 1.0
p.set_facecolors(np.clip(newcolors.reshape(flat_shape, 4), 0, 1))
return p
```
but it doesn't look quite as good as your picture, maybe you know best how to fiddle with the contrast, or maybe it's not possible this way and you have to use `
![Screenshot at 2021-11-02 18-36-18](https://user-images.githubusercontent.com/39047984/139916947-fab06f5f-5ce1-4239-b8d3-0dff45aeb52a.png)
imshow` and alpha values?

Reply:
On second thought, it might be just simpler to call the pcolormesh wrapper if there is a color requested, and then overlay an image with imshow and alpha blending?

Reply:
Try this: `newcolors[..., :3] *= .6 + (lic_data_rgba[...,:3] - 0.5) + 1.0` , with lic_data_rgba as computed in the line_integral_convolution wrapper function. You'll get better results due to the better contrast inside the lic, however I don't really think this is a solution because the results will depend quite heavily on the colormap the user requests, which means we'll need to fiddle the values in this line of code to get optimal-looking results for each colormap. Also, should you use any non-sequential colormaps (such as jet for example), you'll get some pretty bad looking plots since we're applying a linear transformation on all the pixel color values.

To circumvent this problem, we need to work with the opacity of the pixels (fourth axis in newcolors and lic_data_rgba), but unfortunately this does seem to cause pixel-edge artifacts as you mentioned. So I agree more with your second thought, it might be easier to simply use imshow with alpha blending should the user request a lic, and pcolormesh for all other purposes.